packages/indexer-common/src/indexer-management/actions.ts

```diff
@@ -49,11 +85,17 @@ export class ActionManager {
     await this.models.Action.sequelize!.transaction(
       { isolationLevel: Transaction.ISOLATION_LEVELS.SERIALIZABLE },
       async (transaction) => {
-        // Execute already approved actions first
-        const approvedActions = await this.models.Action.findAll({
-          where: { status: ActionStatus.APPROVED },
-          transaction,
-          lock: transaction.LOCK.UPDATE,
+        // Execute already approved actions in the order of type and priority
+        const actionTypePriority = ['unallocate', 'reallocate', 'allocate']
```

Comment:
Can we include a quick explanation of the type ordering? i.e why "unallocate" first?